src/operator/controllers/intents_reconcilers/sarama_cluster_admin_matcher.go

```diff
@@ -0,0 +1,76 @@
+package intents_reconcilers
+
+import (
+	"fmt"
+	"github.com/Shopify/sarama"
+	"github.com/golang/mock/gomock"
+	"reflect"
+)
+
+type SaramaResource []*sarama.ResourceAcls
+
+// Not used, required only to implement gomock.Matcher interface
+func (s *SaramaResource) String() string {
+	return fmt.Sprintf("{SaramaResource %p}", s)
+}
+
+func (s *SaramaResource) Matches(arg interface{}) bool {
+	expected := *s
+	actual := reflect.ValueOf(arg).Interface().([]*sarama.ResourceAcls)
+
+	if len(expected) != len(actual) {
+		return false
+	}
+
+	for i := 0; i < len(expected); i++ {
+		if expected[i] == nil && actual[i] == nil {
+			continue
+		} else if expected[i] == nil || actual[i] == nil {
+			return false
+		}
+
+		a := *expected[i]
+		b := *actual[i]
+
+		if a.Resource != b.Resource || len(a.Acls) != len(b.Acls) {
+			return false
+		}
+
+		for i := 0; i < len(a.Acls); i++ {
+			if !reflect.DeepEqual(a.Acls[i], b.Acls[i]) {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+func MatchSaramaResource(s SaramaResource) gomock.Matcher {
+	return &s
+}
+
+type SaramaAclFilter sarama.AclFilter
+
+// Not used, required only to implement gomock.Matcher interface
```

Comment:
This appears in test results. For example, when a test fails and it wants to show what's the comparison it made, it will use this `String()` to print the two sides of the comparison.

Reply:
Implemented